ebu/benchmarkstt
# Basic setup

Things done in this version:
- setup basic structure
- created some normalisers
- created sphinx docs, see https://mikesmitheu.github.io/ai-benchmarking-stt/docs/build/html/index.html
- created a command line tool
- created proper pip setup for easy installation
- created manpage for command line tool
- created jsonrpc api
- added bash completion for command line tool
- added testing via pytest doctesting

todos:
- could use some more cleanup
- maybe add comments to the more complex code, eg. normalisation.name_to_normaliser (?)
- create api frontend
- create unittests
- move/merge md docs into sphinx docs where applicable
- write own config and csv file readers instead of relying on csv.reader to make it more robust (see "known bugs" below
- ...

known bugs, will be fixed with writing a custom own csv reader (didn't do that for now as this is more to show rudimentary structure):
- "#test" is incorrectly seen as a comment, if it is quoted, it should not be seen as a comment
- for config: if ending a line with a space, it will add a '' as last argument, need to pre-trim line before splitting into arguments
